dns/ddclient/src/opnsense/mvc/app/models/OPNsense/DynDNS/DynDNS.xml

```diff
@@ -110,6 +110,14 @@
                     <Required>N</Required>
                     <mask>/^[^\n]*$/</mask>
                 </password>
+                <netcupAPIKey type="UpdateOnlyTextField">
```

Comment:
API key is usually the password and the user name is left blank

Reply:
Just to clarify:
The Netcup DNS API requires a login which then returns an API Session ID. For the login the following input is required:
- customer number
- API password
- API key

The closest match to the existing fields were:
- customer number => username
- API password => password
- API key => no existing field, hence the introduction of a new field

It would also be possible to do something like this:
Enter both, API key and API password into the existing password field, separated with e.g. colon:
username field: <API key>:<API password>
The code could then split(':') the password field and get API key and password from it. However, this would require some documentation somewhere.

Reply:
> Enter both, API key and API password into the existing password field

That would be my preference as well. Docs are here and a special note for cloudflare is there already: https://docs.opnsense.org/manual/dynamic_dns.html#provider-specific-configuration

If the API key has a fixed length input without separator would be possible as well.